### ISSUE TITLE
Careful when training using Alpaca format! You may be training WRONG.

### ISSUE THREAD

**Issue by FartyPants** [oobabooga/text-generation-webui#3862]
### Describe the bug

Summary

Llama tokenizer make different tokens if you use ### or \n### and that makes all the difference in interference

Details
This is to inform anybody when you are training.
I spent some time debugging this - because my training somehow sucked when I used ### Instruction ### Response, the interference was all wrong

Short story:
If you don't prepend \n before ### instructions, the token will be different than if you put \n before it
as there is token 835 that is " ###" and 2277, 29937 that is ## and #

![image](https://github.com/oobabooga/text-generation-webui/assets/23346289/3e4566ea-62b3-4b75-a92c-2483102d9398)

The consequences are obvious: when you haphazardly mixing token 835 vs 2277, 29937 

So your training MAY be wrong if you have [835] ### Instructions but [2277, 29937] ### Response and obviously the result LORA will behave erratically.

(I won't even comment on the "▁": 29871 whitespace token that always gets mangled there)

There is nothing to fix here, it's a BUG in the system - just make sure you are aware of this!

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

training where ### Instructions is the first thing in the format 

### Logs

```shell
no logs
```

### System Info

```shell
Windows
```

**Comment by hypersniper05** [comment]
Thank you, What about the padding? I've read it is also recommended to use right padding instead of left. `padding_side=right`

**Comment by Maki9009** [comment]
can i ask what the current format is? because its broken for me it wont let me train 
keeps saying key mismatch

**Comment by FartyPants** [comment]
> Thank you, What about the padding? I've read it is also recommended to use right padding instead of left. `padding_side=right`

I don't think it makes any difference in PEFT, but it may be interesting to try. Where did you read about the right being recommended?

**Comment by Maki9009** [comment]
> > Thank you, What about the padding? I've read it is also recommended to use right padding instead of left. `padding_side=right`
> 
> I don't think it makes any difference in PEFT, but i'd be willing to try. Where did you read about the right being recommended?

I im wondering how does you Alpaca formate looks like? could you provide me with an example? since iv been train my dataset using it and i keep getting errors that its wrong / doesnt match
